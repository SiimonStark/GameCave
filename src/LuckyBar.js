import React, {Component} from 'react';

export default class LuckyBar extends Component {
  handleLucky = () => {
    this.props.updateState({isLucky: this.props.isLucky ? false : true});
  }
  render() {
    let buttonSelector = this.props.isLucky ? "button lucky current" : "button lucky";
    return(
      <div className="lucky-bar search-section">
        <button className={buttonSelector} onClick={this.handleLucky}>RANDOM</button>
      </div>
    )
  }
}