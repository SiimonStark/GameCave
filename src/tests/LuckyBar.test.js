import React from 'react';
import ReactDOM from 'react-dom';
import LuckyBar from '../LuckyBar';
import { shallow } from 'enzyme';

const updateStateMock = jest.fn();

describe('LuckyBar', () => {
  let wrapper;
  beforeEach(() => {
    wrapper = shallow(
      <LuckyBar 
      updateState={updateStateMock}/>
    );
  });

  it('should match snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  });

  it('should have default state', () => {
    expect(wrapper.state()).toEqual({
      active: false
    });
  })

  it('should invoke updateState when button is clicked', () => {
    wrapper.find('.button').simulate('click');
    expect(updateStateMock).toBeCalled();
  })

  it('should update active state on button click', () => {
    expect(wrapper.state('active')).toEqual(false);
    wrapper.find('.button').simulate('click');
    expect(wrapper.state('active')).toEqual(true);
  })
})