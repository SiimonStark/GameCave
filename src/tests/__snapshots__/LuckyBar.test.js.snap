// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LuckyBar should match snapshot 1`] = `
<div
  className="lucky-bar search-section"
>
  <button
    className="button lucky"
    onClick={[Function]}
  >
    RANDOM
  </button>
</div>
`;
